### ISSUE TITLE
Elastic agents fails to spin up in EKS version 1.29

### ISSUE THREAD

**Issue by andcri** [gocd/helm-chart#101]
Hello,
I m using the latest version of the chart with the latest elastic agent plugin, everything works fine on my eks 1.25 cluster but the same configuration doesnt seem to work on version 1.29. Specifically the agents are not getting created whenever a job starts and I m getting this logs from the UI

![image](https://github.com/gocd/helm-chart/assets/52462858/cd78ec55-ad92-41db-b19f-03bc129605d7)

From the logs of the server I m getting this:

![image](https://github.com/gocd/helm-chart/assets/52462858/1bdb4f38-39fc-44a4-a768-ec31a16ab51d)
 
but from previous closed issues I saw that this message could be safely ignored.

Is this an incompatibility with k8s versions > 1.25 or is there any extra configuration needed that wasnt required before? Thanks.

**Comment by chadlwilson** [comment]
Can you look at the logs from the elastic agent plugin and in the server log and see what it says? There'll be a big stack trace somewhere, which is what we need to see.

I doubt that message can be "ignored", which specific issue are you referring to?

I'm pretty confident it will have been working with Kubernetes through `1.28` based on users I know of, but `1.29` might be a niggle.

The chart is tested against `1.29` however I believe it might only test with a static agent, not elastic agent and unfortunately [the plugin itself](https://github.com/gocd/kubernetes-elastic-agents) does not have automated tests against real clusters of various versions.

**Comment by andcri** [comment]
How can I get those logs? do I need to just enable them with this extra line in the wrapper conf `wrapper.java.additional.100=-Dplugin.cd.go.contrib.elasticagent.kubernetes.log.level=debug` and they will be visible in the pod logs? thanks.

Regarding the message to be ignored I m referring to this: https://github.com/gocd/kubernetes-elastic-agents/issues/73

Thanks.

**Comment by chadlwilson** [comment]
> How can I get those logs? do I need to just enable them with this extra line in the wrapper conf `wrapper.java.additional.100=-Dplugin.cd.go.contrib.elasticagent.kubernetes.log.level=debug` and they will be visible in the pod logs? thanks.

The plugin specific logs I believe don't feed through to the main container STDOUT. It's annoying in a container environment, and not sure why it has been set up that way historically. probably should investigate changing the default in a container environment.

If you can `exec` into the pod they would be in `/godata/logs/plugin-*.log` or you could `kubectl cp` them out from there to somewhere locally. Same as mentioned at https://github.com/gocd/kubernetes-elastic-agents/issues/73#issuecomment-502624385

> Regarding the message to be ignored I m referring to this: [gocd/kubernetes-elastic-agents#73](https://github.com/gocd/kubernetes-elastic-agents/issues/73)

It's only the plugin settings request failure that was "safe to ignore" on the old plugin version, not the failure to create an agent :-)

**Comment by chadlwilson** [comment]
Hmm, that is less informative than usual, but looks like it could be a problem with network connectivity or timeouts within your particular EKS setup: https://github.com/fabric8io/kubernetes-client/issues/5707 It implies the client is cancelling the request after a timeout (default is `10s`).

I have no idea why it would be so slow in your environment (10s is a very long time for an API like this call), but you could increasing the Cluster Request Timeout in the cluster profile settings and see if the problem persists?

**Comment by andcri** [comment]
It was the timeout 😂 , I increased it and now the agent gets spinned up, thanks so much for the suggestion :)

**Comment by chadlwilson** [comment]
Yeah I'd love to have some idea too. I'd have thought this was just an asynchronous API call but haven't dug into see whether admission controllers need to fire first to get response - so if there are changes to pod admission in later versions or a problem there that's one thing that occurs to me.

**Comment by andcri** [comment]
I ll keep you posted on that once I found out 👍
